src/XMakeBuildEngine/Definition/ToolsetConfigurationReader.cs

```diff
@@ -223,50 +192,50 @@ protected override IEnumerable<ToolsetPropertyDefinition> GetSubToolsetPropertyD
         /// </summary>
-        protected override Dictionary<string, List<string>> GetMSBuildExtensionPathsSearchPathsTable(string toolsVersion, string os)
+        protected override Dictionary<string, List<string>> GetProjectImportSearchPathsTable(string toolsVersion, string os)
         {
             Dictionary<string, List<string>> kindToPathsCache;
             var key = toolsVersion + ":" + os;
-            if (_kindToPathsCachePerToolsVersion.TryGetValue(key, out kindToPathsCache))
+            if (_projectImportSearchPathsCache.TryGetValue(key, out kindToPathsCache))
             {
                 return kindToPathsCache;
             }
 
             // Read and populate the map
             kindToPathsCache = new Dictionary<string, List<string>>();
-            _kindToPathsCachePerToolsVersion[key] = kindToPathsCache;
+            _projectImportSearchPathsCache[key] = kindToPathsCache;
 
             ToolsetElement toolsetElement = ConfigurationSection.Toolsets.GetElement(toolsVersion);
-            var propertyCollection = toolsetElement?.AllMSBuildExtensionPathsSearchPaths?.GetElement(os)?.PropertyElements;
+            var propertyCollection = toolsetElement?.AllProjectImportSearchPaths?.GetElement(os)?.PropertyElements;
             if (propertyCollection == null || propertyCollection.Count == 0)
             {
                 return kindToPathsCache;
             }
 
-            kindToPathsCache = ComputeDistinctListOfFallbackPathsFor(propertyCollection);
+            kindToPathsCache = ComputeDistinctListOfSearchPaths(propertyCollection);
 
             return kindToPathsCache;
         }
 
         /// <summary>
-        /// Returns a list of the search paths for a given MSBuildExtensionsPathReferenceKind
+        /// Returns a list of the search paths for a given search path property collection
         /// </summary>
-        private Dictionary<string, List<string>> ComputeDistinctListOfFallbackPathsFor(ToolsetElement.PropertyElementCollection propertyCollection)
+        private Dictionary<string, List<string>> ComputeDistinctListOfSearchPaths(ToolsetElement.PropertyElementCollection propertyCollection)
         {
             var pathsTable = new Dictionary<string, List<string>>();
 
             foreach (ToolsetElement.PropertyElement property in propertyCollection)
             {
-                //FIXME: handle ; in path on Unix
-                var kind = MSBuildExtensionsPathReferenceKind.FindIn($"$({property.Name})");
-                if (kind.Equals(MSBuildExtensionsPathReferenceKind.None))
+                if (string.IsNullOrEmpty(property.Value) || string.IsNullOrEmpty(property.Name))
                 {
                     continue;
                 }
 
                 var paths = new List<string>();
+
+                //FIXME: handle ; in path on Unix
                 foreach (var extnPath in property.Value.Split(new [] { _separatorForExtensionsPathSearchPaths }, StringSplitOptions.RemoveEmptyEntries))
```

Comment:
should have seen this before, but this could just be:
```C#
var paths =
    property.Value.Split(new[] {_separatorForExtensionsPathSearchPaths},
        StringSplitOptions.RemoveEmptyEntries).Distinct();
```
right?

Do we need/want to worry about case consolidation for Windows (and default OS X HFS+)?